CONTRIBUTING.md

```diff
@@ -5,20 +5,91 @@
+* `feature` is increased when set of exported definitions is strictly augmented.
+* `breaking` is increased when breaking changes are made.
+* `marketing` is increased for releases which are important from marketing perspective.
+
+### Code modification policy
+
+**tl;dr**
+
+1. Issues with labels `type:not-code` and `type:fix` can be resolved at any time.
+2. Issues with labels `type:feature` and `type:breaking` can be resolved only after being added to a milestone.
+3. `features` releases happen at most once per two weeks.
+3. `breaking` releases happen at most once per five weeks.
+
+**Actual policy**
+
+We think that a custom prelude shouldn't change often.
+Because of that we have a semi-formal policy describing which modifications can be done to `universum` under which circumstances.
+We split all issues into four semi-formally defined categories:
+1. `not-code`.
+Everything that doesn't modify source code (except non-Haddock comments), `.cabal` files, other files considered by GHC (if they exist).
+Examples: fix a typo in README.md, fix something in CI configuration files, add a pull request template.
+2. `fix`.
+Everything that doesn't fit in `not-code`, but doesn't change set of exported definitions and their types or semantics.
+Examples: fix compatibility with a new version of a library/compiler, improve performance of some function (keeping its semantics intact).
+Note: despite the name, issues of this type do not necessary correspond to bug fixes.
+3. `feature`.
+Modifications which make set of exported definitions a superset of the old set of exported definitions.
+I. e. all existing definitions are left intact, only something new is added.
+Example: add a new function, re-export a type class, add a new instance of some type class.
+4. `breaking`.
+Basically all other changes.
+Examples: remove of a function, rename a type class, change arguments order of a function.
+
+Each issue should belong to only one category.
+If an issue involves multiple changes of different types, it should be split into smaller issues
+
+For each category we have a label on GitHub, these labels start with the `type:` prefix.
+There is also the `type:unclear` label for issues which don't have a list of specific actions to be done.
+
+Processing of an issue depends on its category. Specifically:
+1. A `not-code` issue can be resolved at any time.
+PRs can be merged whenever GitHub allows it.
+Minimal number of approvals is set to 2 in GitHub settings.
+2. Rules for the `fix` issues are the same as for `not-code` with an additional rule that PRs for such issues should increase the last component in the version and this version should be uploaded to Hackage as soon as the PR is merged (also new tag and release are created).
+3. Work on the `feature` issues is organized in milestones.
+Each milestone corresponds to a release which increases at least the third (`feature`) version component.
```

Comment:
I wonder who manages milestones. Is it the same person who creates releases and uploads them to hackage?

Reply:
Yes, I call these people _maintainers_ and that's basically what I mean by line below:
>  Scope of each milestone is discussed in issues and is set by maintainers.